src/renderer/src/routes/modals/Notifications/functionByNotificationActionType.tsx

```diff
@@ -0,0 +1,45 @@
+import { useNavigate } from 'react-router-dom'
+import { AccountHelper } from '@renderer/helpers/AccountHelper'
+import { selectAccounts } from '@renderer/hooks/useAccountSelector'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
+import { RootStore } from '@renderer/store/RootStore'
+import { TAccountHelperPredicateParams } from '@shared/@types/helpers'
+import { TNotificationAction } from '@shared/@types/store'
+import { match } from 'ts-pattern'
+
+type TFunctionParams<T> = {
+  modalActions: ReturnType<typeof useModalNavigate>
+  pageNavigate: ReturnType<typeof useNavigate>
+  notificationAction: T
+}
+
+type TFunctionByNotificationActionType = {
+  [K in TNotificationAction['type']]: (params: TFunctionParams<TNotificationAction & { type: K }>) => Promise<void>
+}
+
+const getAccount = (predicate: TAccountHelperPredicateParams) => {
+  const state = RootStore.store.getState()
+  const accounts = selectAccounts(state)
+  const account = accounts.find(AccountHelper.predicate(predicate))
+  return account
+}
+
+export const functionByNotificationActionType: TFunctionByNotificationActionType = {
```

Comment:
What do you think?

```typescript
export const notificationActionTypes: TNotificationActionTypes = { ... }
```

I don't like so much the "function" in the variable name.

Reply:
`notificationActionTypes` is not what this variable is, with that name, I would expect it to be the type and not functions that perform the action